gc/base/standard/ParallelSweepScheme.cpp

```diff
@@ -509,7 +509,7 @@ MM_ParallelSweepScheme::sweepChunk(MM_EnvironmentBase *env, MM_ParallelSweepChun
 	uintptr_t darkMatterBytes = 0;
 	uintptr_t darkMatterCandidates = 0;
 	uintptr_t darkMatterSamples = 0;
-	const UDATA darkMatterSampleRate = (0 == _extensions->darkMatterSampleRate)?UDATA_MAX:_extensions->darkMatterSampleRate;
+	const UDATA darkMatterSampleRate = ((0 == _extensions->darkMatterSampleRate) || (_extensions->usingSATBBarrier())) ? UDATA_MAX:_extensions->darkMatterSampleRate;
```

Comment:
is sampling going to work with this mark-last-object or it's a different problem!?

Reply:
Nope, it will not work, it's a different problem. Sampling iterates using `MM_HeapMapWordIterator markedObjectIterator` which doesn't like batch marked tlhs